### PR TITLE
feat: add support Google Gemini API in LLMs

### LINKED ISSUE #879: Add Support for Google Gemini pro api
### 🚀 The feature

Add Support for Google Gemini pro api

### Motivation, pitch

Gemini pro has a great potential 

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dudesparsh**
Hi @gventuri,

This PR aims at adding support for Google Gemini API (pro models) in LLMs.

- [x] Closes #879 (Replace xxxx with the GitHub issue number).
- [x] Tests added and passed if fixing a bug or adding a new feature.
- [x] All [code checks passed](https://github.com/gventuri/pandas-ai/blob/main/CONTRIBUTING.md#-testing).

## Summary by CodeRabbit

- **New Features**
	- Introduced a method for data summarization to enhance data analysis capabilities.
	- Added support for the Google Gemini API, enabling advanced language model generation.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -41,6 +40,7 @@ yfinance = { version = "^0.2.28", optional = true }
 sqlalchemy-databricks = { version = "^0.2.0", optional = true }
 snowflake-sqlalchemy = { version = "^1.5.0", optional = true }
 pillow = "^10.1.0"
+google-generativeai = "^0.3.2"
```
Comment (gventuri) [inline pyproject.toml:43]:
```suggestion
google-generativeai = {version = "^0.3.2", optional = true}
```

**Comment by dudesparsh** [comment]
@gventuri  - Unit tests should be added for the Gemini API - do we intend to add those? ( If yes, I can take it up )

**Comment by gventuri** [comment]
@dudesparsh sure, increasing the coverage is a key priority. Would love if you could work on it :)

**Comment by dudesparsh** [comment]
Hi,

@gventuri I wrote some unit tests for the Gemini API. However a few of the tests are failing, can you or someone assist me in updating these tests? 

Also since those tests are failing, I think I am not able to push those tests. Can you suggest what might be a good way to commit those changes to my local branch, such that they can be reviewed?

Thanks.

**Comment by gventuri** [comment]
Hi @dudesparsh, thanks a lot! Sounds great, glad to help with the tests. 

I suggest you push the changes with `git commit -m 'Your message' --no-verify`. This should skip the precommit command and commit the change even if tests are failing.

As an alternative, you can comment out the tests or skip them.

**Pull request merged**